routers/usersRouter.js

```diff
@@ -0,0 +1,95 @@
+const usersDb = require("../data/helpers/usersDb");
+const bcrypt = require("bcryptjs");
+const customMw = require("../customMiddleware");
+
+const express = require("express");
+const router = express.Router();
+const protect = customMw.protect;
+const auth = customMw.auth;
+
+// router.use("/restricted", auth())
+
```

Comment:
Really like that, you're checking for correct data types, and the absence of data as well 🔥 Check out this [article](https://medium.freecodecamp.org/how-to-make-input-validation-simple-and-clean-in-your-express-js-app-ea9b5ff5a8a7) about input validation in express. There many ways to get this done, but hopefully this will get you thinking about how you can do this with middleware 👍